simpleviewinc/re-theme
# Zen 351 Clean Up

**Ticket**: [ZEN-351](https://jira.simpleviewtools.com/browse/ZEN-351)

## Context
* events force is defining their styles in CSS
* their clients can define custom CSS as well
* currently, the tap, using re-theme, is only compatible with JSON style objects
* we need to come up with a solution to interfacing with CSS
* it also needs to be able to update styles if the css updates

## Goal

* Needed to clean up some items with re-theme
* This is to prepare the repo, for the future changes down the road
* There's a few small performance updates

## Updates

* Added - `container folder`
  * Allow running the ReTheme app in a container
  * Use the keg-cli to run as an `injected-app`
* `configs/rollup.config.js`
  * Update rollup build to not export a `.web.js` and a `.native.js`
  * Now exports a `.native.js` and `.js`
* Moved config files into a config folder
* Updated `jsutils` to the latest version from npm
* Removed all alias from the build and config
  * No longer required with the latest update
* `src/dimensions/dimensions.js`
  * Added a wrapper around the dimensions object to show a warning when it's called without being initialized
* `src/helpers/getTheme.js`
  * Updated to use the event system, to pull in the theme
  * This way there's no more binding the `get` method to the theme
  * Which was creating a new get method every time `useTheme` was called
* `src/dimensions/webDimensions.js`
  * Extracted `hasDomAccess ` to it's own file so it can be re-used
* Added a `useStylesCallback` hook for building dynamic styles
  * See the the [keg-core pull request](https://github.com/simpleviewinc/keg-core/pull/63) for an example of using it
* `src/hooks/useTheme.js`
  * Removed the binding of the `getTheme` helper
* `src/theme/buildTheme.js`
  * Added the `getTheme` and `joinTheme` helpers here instead of in the `useTheme` hook
* `src/theme/manageTheme.js`
  * Added to hold the current theme  whenever it gets re-built
  * Allows the other helper methods to get access to it when needed

## Testing

**Test 1**
* Run the docker package to test the tap =>
  * `keg dpg run --package docker.pkg.github.com/simpleviewinc/keg-packages/tap:zen-290-use-expo-vector-icons --command tap:start`
  * Ensure everything works as expected

**Test 2**
* Run the docker package to test keg-core =>
  * `keg dpg run --package docker.pkg.github.com/simpleviewinc/keg-packages/keg-core:usestylecallback-example`
  * Ensure everything works as expected

**Test 3**
* Run the docker package to test re-theme app =>
  * First you need to link your local retheme as a tap to the keg cli
    * `cd /path/to/retheme/folder`
    * `keg tap link retheme`
 * Then run the package cmd below 
  * `keg retheme pack run --package docker.pkg.github.com/simpleviewinc/keg-packages/retheme:zen-351-clean-up-builds`
  * Navigate to `retheme.kegdev.xyz` in your browser
  * Ensure everything works as expected

**Test 4**
* Connect to the running `Re-theme` container
  * `docker exec -it package-retheme-zen-351-clean-up-builds /bin/bash`
* Run `yarn test`
  * Ensure all tests pass
